101-webapp-custom-deployment-slots/azuredeploy.json

```diff
@@ -0,0 +1,89 @@
+{
+  "$schema": "http://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "baseResourceName": {
+      "defaultValue": "GEN-UNIQUE-12",
+      "type": "string",
+      "metadata": {
+        "description": "Name of the resource"
+      },
+      "maxLength": 15
+    },
+    "environments": {
+      "defaultValue": [
+        "Dev",
+        "QA01",
+        "QA02",
+        "Preview"
+      ],
+      "type": "array",
+      "metadata": {
+        "description": "Array with the names for the environment slots"
+      },
+      "maxLength": 19
+    }
+  },
+
+  "variables": {
+    "webAppPortalName": "[concat(parameters('baseResourceName'), 'Portal')]",
+    "appServicePlanName": "[concat('AppServicePlan-', parameters('baseResourceName'))]"
+  },
+
+  "resources": [
+    {
+      "apiVersion": "2015-08-01",
+      "type": "Microsoft.Web/serverfarms",
+      "kind": "app",
+      "name": "[variables('appServicePlanName')]",
+      "location": "[resourceGroup().location]",
+      "comments": "This app service plan is used for the web app and slots.",
+      "tags": {
+        "displayName": "AppServicePlan"
+      },
+      "properties": {},
+      "dependsOn": [],
+      "sku": {
+        "name": "S1"
```

Comment:
since this has a limit of 5 slots, we need to fix the param maxLength above...   You could just set this based on the size of the array passed in using an if() stmt

Reply:
++
Done. That's a great idea!